lib/_stream_writable.js

```diff
@@ -498,22 +502,42 @@ function onwrite(stream, er) {
     }
 
     if (sync) {
-      process.nextTick(afterWrite, stream, state, cb);
+      // It is a common case that the callback passed to .write() is always
+      // the same. In that case, we do not schedule a new nextTick(), but rather
+      // just increase a counter, to improve performance and avoid memory
+      // allocations.
+      if (state.afterWriteTickInfo !== null &&
+          state.afterWriteTickInfo.cb === cb) {
+        state.afterWriteTickInfo.count++;
+      } else {
+        state.afterWriteTickInfo = { count: 1, cb, stream, state };
+        process.nextTick(afterWriteTick, state.afterWriteTickInfo);
```

Comment:
Is there any difference in just using `afterWrite` directly here?

Reply:
@mscdex We need to allocate an object anyway so that we can modify `count` later, so that’s why it’s not just spreading the arguments right now